### PR TITLE
Serviceaccounts: create serviceaccount api

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
Currently there is no way to create a serviceaccount without creating a apikey.
This PR implements a endpoint from which serviceaccounts will be created using the endpoint.

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana-enterprise/issues/2276

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/serviceaccounts/database/database.go [hunk 1]
```diff
@@ -18,6 +20,23 @@ func NewServiceAccountsStore(store *sqlstore.SQLStore) *ServiceAccountsStoreImpl
 	}
 }
 
+func (s *ServiceAccountsStoreImpl) CreateServiceAccount(ctx context.Context, sa *serviceaccounts.CreateServiceaccountForm) (user *models.User, err error) {
+	// create a new service account - "user" with empty permissions
+	cmd := models.CreateUserCommand{
+		Login:            "Service-Account-" + uuid.New().String(),
+		Email:            uuid.New().String(),
+		Password:         "Password-" + uuid.New().String(),
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/database/database.go:28]:
```suggestion
```
nit: should probably remove the need for a email and password for a serviceaccount

**Review by donomii — CHANGES_REQUESTED** [review]
I'm against using definitions for well-known constants.  The HTTP response codes are an international standard that have been implemented millions of times across the globe.  Giving them new names is more confusing than just typing the number.  At least I can google the number if I don't know it.

They are never going to change, so the traditional reason for naming numbers does not exist.

File: pkg/services/serviceaccounts/api/api.go [hunk 2]
```diff
@@ -40,9 +43,27 @@ func (api *ServiceAccountsAPI) RegisterAPIEndpoints(
...
+	}
+	user, err := api.service.CreateServiceAccount(c.Req.Context(), &cmd)
+	switch {
+	case errors.Is(err, serviceaccounts.ErrServiceAccountNotFound):
+		return response.Error(http.StatusBadRequest, "Failed to create role with the provided name", err)
+	case err != nil:
+		return response.Error(http.StatusInternalServerError, "Failed to create service account", err)
+	}
+
+	return response.JSON(http.StatusCreated, user).SetHeader("Location", fmt.Sprintf("/api/serviceaccounts/%d", user.Id))
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/api/api.go:64]: @jeremy-watkins are you referring to this?

I am not sure about your comment around making a new name for it. As I get 201 Created.
Maybe you are referring to something else here


we are also doing this in the access control package in enterprise

Comment (donomii) [inline pkg/services/serviceaccounts/api/api.go:64]:
```suggestion
	return response.JSON(http.StatusCreated, user)
```

This is a REST endpoint, what is the purpose of setting "Location"?

Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go:64]: I really have no idea, more than we do it in access control.

Comment (donomii) [reply pkg/services/serviceaccounts/api/api.go:64]: wrong Jeremy :D

Yes, I thought that the comment would be attached to the code.  Odd.

Not the returned name, the name in the code.  I don't see the benefit of writing "http.StatusCreated" when you could just write 201.

It's shorter, it's well know, and it's a constant.

We're not in the accesscontrol package, we don't have to follow what they do.

Comment (donomii) [reply pkg/services/serviceaccounts/api/api.go:64]: If we can't clearly state the reason for doing something, we shouldn't do it.

**Review by donomii — APPROVED** [review]
LGTM

**Pull request merged**